### PR TITLE
Fixes #14106 - Compress 15-30% (on avg) controller responses with Rack::Deflater

### PR DESCRIPTION
**Pull request by dLobatog**
Using Rack::Deflater we can compress our responses for HTML and assets
using gzip automatically. This allows us to serve considerably smaller
(in kb) pages, which should go over the wire faster as well.

Trying out /login, /hosts, and /dashboard, processing times were similar
with and without this, but the page size was reduced up to 80% in some
cases.

See the following gist with benchmarks using siege:
https://gist.github.com/dLobatog/600f151fd0676345c95a - notice some
pages like /login are reduced a whooping 75% in size.

### CODE REVIEW

File: config.ru [hunk 1]
```diff
@@ -4,5 +4,6 @@ require ::File.expand_path('../config/environment',  __FILE__)
 # apply a prefix to the application, if one is defined
 # e.g. http://some.server.com/prefix where '/prefix' is defined by env variable
 map ENV['RAILS_RELATIVE_URL_ROOT'] || '/'  do
+  use Rack::Deflater
```
Comment (ohadlevy) [inline config.ru:7]: does this also applies to assets?

Comment (ohadlevy) [inline config.ru:7]: does your benchmark is on prod or dev env?

Comment (dLobatog) [inline config.ru:7]: Looks like I didn't pay enough attention and just ran 'passenger start' w/o RAILS_ENV=production, I'll reopen if I can figure this out better but at first glance I cannot reproduce these benchmarks on production :/

**Comment by lzap** [comment]
I don't like compressing content with Ruby when we can do the same with Apache httpd consuming less resources. It's just a matter of configuration I guess, we already serve CSS/JS in deflated form via Apache.

**Pull request closed without merging**